extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/FileService.java

```diff
@@ -1353,11 +1353,15 @@ public RepositoryFileTreeDto doGetTree( String pathId, Integer depth, String fil
     repositoryRequest.setIncludeAcls( includeAcls );
     repositoryRequest.setIncludeSystemFolders( includeSystemFolders );
 
-    RepositoryFileTreeDto tree = getRepoWs().getTreeFromRequest( repositoryRequest );
-
+    RepositoryFileTreeDto tree = null;
+    try {
+      tree = getRepoWs().getTreeFromRequest( repositoryRequest );
+    } catch ( Exception e ) {
+      logger.error( "Problem getting path: " + path );
```

Comment:
This catch clause swallows any exception that might occur here.
I'm guessing that we want to catch a folder not found or something related - isn't there a more specific type of exception we can catch here?

At the very least we should log the exception. Right now, that information is lost.

Reply:
I believe so, I'll try to change it for a more specific exception.